lib/ansible/cli/__init__.py

```diff
@@ -407,7 +408,8 @@ def unfrack_paths(option, opt, value, parser):
         else:
             pass  # FIXME: should we raise options error?
 
-        setattr(parser.values, option.dest, paths)
+        # We reverse the list to use the order provided on the CLI
```

Comment:
I think the callback gets called for each cli option, so for '--roles-path roles/ --roles-path ~/roles --roles-path last_roles/'
this seems like it would reverse all of the accumulated roles-path on each invocation.
ie,
```
--roles roles/                   ['roles/']     
                --reverses-->    ['roles/']
--roles ~/roles                  ['roles/', '/home/adrian/roles/']
                --reverses-->    ['/home/adrian/roles/', 'roles/']
-- roles last_roles              ['/home/adrian/roles/', 'roles/', 'last_oles/']
                --reverses-->    ['last_roles/', 'roles/', '/home/adrian/roles/']
```

Reply:
Though tests seem to pass so I guess it works out...

Reply:
ah that would only apply for the paths after splitting a  value into a list (like --roles-path foo/:bar/:/tmp/roles)

Reply:
I believe the change I just pushed handles this